src/main/java/org/jboss/config/LotteryConfig.java

```diff
@@ -10,35 +10,64 @@
 import io.quarkus.runtime.annotations.RegisterForReflection;
 
 import java.time.Duration;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.TreeSet;
 
 // Used for deserializing the config file
 @RegisterForReflection(ignoreNested = false)
 public record LotteryConfig(
         @JsonProperty(required = true) Duration delay,
-        List<Participant> participants) {
+        List<Participant> participants,
+        @JsonProperty(required = true) @JsonDeserialize(as = HashMap.class) Map<String, Set<String>> availableProjects) {
 
     public static final String FILE_NAME = "jira-issue-lottery.yml";
 
     public record Participant(
             @JsonProperty(required = true) String email,
             Set<Project> projects) {
+        public record Participation(@JsonProperty(required = true) int maxIssues) {
+        }
+    }
+
+    public record Project(
+            @JsonProperty(required = true) String project,
+            @JsonDeserialize(as = TreeSet.class) Set<String> components,
+            @JsonUnwrapped @JsonProperty(access = JsonProperty.Access.READ_ONLY) Participant.Participation participation) {
+        // https://stackoverflow.com/a/71539100/6692043
+        @JsonCreator
+        public Project(@JsonProperty(required = true) String project, @JsonProperty(required = true) Set<String> components,
+                @JsonProperty(required = true) int maxIssues) {
+            this(project, components, new Participant.Participation(maxIssues));
+        }
+    }
 
-        public record Project(
-                @JsonProperty(required = true) String project,
-                @JsonDeserialize(as = TreeSet.class) Set<String> components,
-                @JsonUnwrapped @JsonProperty(access = JsonProperty.Access.READ_ONLY) Participation participation) {
-            // https://stackoverflow.com/a/71539100/6692043
-            @JsonCreator
-            public Project(@JsonProperty(required = true) String project, @JsonProperty(required = true) Set<String> components,
-                    @JsonProperty(required = true) int maxIssues) {
-                this(project, components, new Participation(maxIssues));
+    public void validate() throws ConfigException {
```

Comment:
I would probably keep the config just for config and validation in a different file or directly in LotteryConfigProducer could have this method as private.